### PR TITLE
Support create index ALGORITHM and LOCK options parse

### PR DESCRIPTION
**Pull request by zihaoAK47**
Refer #25862.

Changes proposed in this pull request:
  - Support create index ALGORITHM and LOCK options parse

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ✓] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ✓] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ✓] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ✓] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by zihaoAK47** [comment]
@TeslaCN Mentor please review.

File: parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLCreateIndexStatement.java [hunk 1]
```diff
@@ -17,11 +17,24 @@
...
+import java.util.LinkedList;
+
 /**
  * MySQL create index statement.
  */
+@Setter
+@Getter
 public final class MySQLCreateIndexStatement extends CreateIndexStatement implements MySQLStatement {
+
+    private final Collection<AlgorithmTypeSegment> algorithmSegment = new LinkedList<>();
```
Comment (TeslaCN) [inline parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLCreateIndexStatement.java:37]: When will there be multiple algorithm options?

**Review by TeslaCN — COMMENTED** [review]
Please add some test cases for your changes.

File: parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLCreateIndexStatement.java (hunk 1 above)
Comment (zihaoAK47) [reply parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLCreateIndexStatement.java:37]: CREATE [UNIQUE | FULLTEXT | SPATIAL] INDEX index_name
    [index_type]
    ON tbl_name (key_part,...)
    [index_option]
    [algorithm_option | lock_option] ...

Sorry, I misunderstood the rules of MySQL syntax and thought it was a multi choice support. 
`[algorithm_option | lock_option] ...`

File: parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/DDLStatement.g4 [hunk 2]
```diff
@@ -207,7 +207,7 @@ truncateTable
 
 createIndex
     : CREATE createIndexSpecification? INDEX indexName indexTypeClause? ON tableName keyListWithExpression indexOption?
-    (alterAlgorithmOption | alterLockOption)*
+    alterAlgorithmOption? alterLockOption?
```
Comment (TeslaCN) [inline parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/DDLStatement.g4:210]: Please refer to MySQL syntax.
https://github.com/mysql/mysql-server/blob/1bfe02bdad6604d54913c62614bde57a055c8332/sql/sql_yacc.yy#L9089

**Comment by zihaoAK47** [comment]
@TeslaCN Tutor, the inspection failed, Here are the reasons for the failure

Error:  Failed to execute goal on project shardingsphere-test-e2e-sql: Could not resolve dependencies for project org.apache.shardingsphere:shardingsphere-test-e2e-sql:jar:5.3.3-SNAPSHOT: Could not find artifact org.apache.shardingsphere:shardingsphere-test-e2e-env:jar:tests:5.3.3-SNAPSHOT in apache.snapshots (https://repository.apache.org/snapshots) -> [Help 1]

**Comment by zihaoAK47** [comment]
> > @TeslaCN导师，我有一个问题。我想使用已经在创建索引任务中编写的代码，例如 algorithmOptionAndLockOption 规则和 ExpectedAlgorithmClause ExpectedLockTypeClause 类。但是，这段代码目前在创建索引分支中。我是否需要在 drop index 分支中重写这段代码？吃惊
> 
> 在这个 PR 合并之后，你可以将 master 分支合并到那个分支中。

I understand. Thank you, mentor

**Review by TeslaCN — APPROVED** [review]

**Pull request merged**